tests/integration/test_compute.py

```diff
@@ -0,0 +1,68 @@
+from fastapi import params
+import numpy as np
+from numpy.testing import assert_array_equal
+import pandas as pd
+
+
+OBSERVATION = (
+    "FOPR",
+    {"values": [1, 2, 3], "errors": [0.1, 0.2, 0.3], "x_axis": ["C", "E", "H"]},
+)
+
+
+def test_univariate_misfits_with_labels(client, create_experiment, create_ensemble):
+    experiment_id = create_experiment("test_ensembles")
+    ensemble_id = create_ensemble(experiment_id=experiment_id)
+    # post observation
+    name, obs = OBSERVATION
+    obs_id = client.post(
+        f"/experiments/{experiment_id}/observations",
+        json=dict(
+            name=name,
+            values=obs["values"],
+            errors=obs["errors"],
+            x_axis=obs["x_axis"],
+        ),
+    ).json()["id"]
+
+    # 10 realizations of 8 values each
+    matrices = np.random.rand(10, 8)
+
+    data_df = {
+        id_real: pd.DataFrame(
+            matrix,
+            columns=[f"{id_real}"],
+            index=["A", "B", "C", "D", "E", "F", "G", "H"],
+        )
+        for id_real, matrix in enumerate(matrices)
+    }
+    for id_real in data_df:
+        resp = client.post(
+            f"/ensembles/{ensemble_id}/records/FOPR/matrix",
+            data=data_df[id_real].to_csv().encode(),
+            headers={"content-type": "application/x-dataframe"},
+            params=dict(realization_index=id_real),
+        )
+        client.post(
+            f"/ensembles/{ensemble_id}/records/FOPR/observations",
+            json=[obs_id],
+            params=dict(realization_index=id_real),
+        )
+
+    misfits = client.get(
+        "/compute/univariate_misfits",
+        params=dict(ensemble_id=str(ensemble_id), response_name="FOPR"),
+    ).json()
+    misfits_df = pd.DataFrame(data=misfits)
+    assert_array_equal(misfits_df.index, obs["x_axis"])
+    assert misfits_df.shape == (3, 10)
```

Comment:
We should probably be asserting the values themselves, not only the shape.

Reply:
Now it's there. See above comment.